### PR TITLE
Helm chart for Dex

### PR DESCRIPTION
**Pull request by keyolk**
Write Basic Dex Deployments.
Including simple client app.
TLS assets must be set.
Issuer must be set.
RedirectURI must be set.

Support GitHub and Google Connector
Use NodePort by default.

### CODE REVIEW

**Comment by coffeepac** [comment]
verify

**Comment by coffeepac** [comment]
ok to test

**Comment by coffeepac** [comment]
add to whitelist

File: dex/values.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
...VE4QU1JSUJDZ0tDQVFFQXAwNVR3b1BFNXUrQkxnQkxqY1VwWkhrcm5FRGgKc3loMUlQaU1iM3lzd09WSkVHOTE5eTFrVzlxNkVwSUpaYURac0JvRU9ERjNBQ2FyZFNyM0djTGZVOWZUS1J1VQplNThCK0doSS94WHVmVzJzTG9DY2RpbDJIREFMSFZWNW1TNWY3ZkVIazFadC9yKzczWXM2R25FcE9QQVoyUENWCkU5RGtVYStHVlB1SDZCUjdMalFYSnlqaGNEOFJIYW0vNHpNblFPMjZGNzRBZ1ZwYzd2eEpremFPcFhtT0Yram0KYnFtYnZIeWk5ZXN5aWowVk9VOWRoZjhVZE9zUFNBV095bENmRFc3RVBwR1VNVVJSTUlEWHZTSzFHZ2tNMlo2MgppL2ZiaXV6YnYxbDRoQmVaTW83dzN5ZDN6MXllTWhIYkNTRVd0MGVCT0pUTk5UaFhJK2lmRTRjZ0F3SURBUUFCCm8xNHdYREFPQmdOVkhROEJBZjhFQkFNQ0FRWXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QWRCZ05WSFE0RUZnUVUKdVhzL09tWk9LZks2UVNHODc2TUtyYlBOSFpZd0dnWURWUjBSQkJNd0VZSVBkbUZuY21GdWRDOXdhMmt2WkdWNApNQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUJBUUFxSEhKek4xK0hyOUNSanJqNnQvczJOYTBlcXBzUFBvbzFuZ0p0CktaWFlOVXNPYzVSbDdLVXpIQXhJamhyaDIwbUFCcEFLTHBVZzNHQ0hwUjRXVTY5K0dyalBFVUpJbmRneit5MVMKS0VNTTEwZWVaQzlxQVRaVVZxdlU0K0ZZM2lBREhibnpKL1ArNU5jdHI4eVRUWlh2MHVwTHdVVHI2aXpVSkluVwpxUjdhbUtuaWZKYU1KZGxpZGQzQVlmYWFtdjlRSkRQU3Z4TWJiOUFha0tzdE94YU1JZWlndFV1MENsK0VVQmpjCmU0V2pKVHdxd1lMc21QNmVtZVgxSHBid0xMTU1OVWRJT29ocm1QYk1KbE5YMUF4aHZOQ05aT21XSDlEbXgrMWMKQWYyeFVZWFlPeUZnV3ZBTzdPZEY2V3UvU3Y3K0FXTFI3QVJJd0VjMXpWcnRScTE1Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0="
```
Comment (coffeepac) [inline dex/values.yaml]: this is a public repo.  do these credentials matter?

File: dex/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+description: A CoreOS Dex chart for Kubernetes
+name: dex
+version: 0.1.0
+keywords: [dex]
```
Comment (coffeepac) [inline dex/Chart.yaml]: Dex is an openId provider, right?  can you a keyword for that.  and any other relevant search terms.

File: dex/values.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      ClientSecret: "MjYzN2Y3OTMyMTNlMmI4ZDZlM2RlM2Y0YWU2NzUzMTVhOWMzZWVkMQ=="
+    Google:
+      ClientId: "ODc2MDc2NjM5NDQ0LW8zdTIzZjY5ZGlsdXMxYTBkNzd0ZHQxb3FzMWZpaDViLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29t"
+      ClientSecret: "dUVJa1lBMEVxS3dzeEgwdGJMXzY4aUhN"
+  Memory: "200m"
+  Cpu: "512Mi"
+  ServicePort: 443
+  Replicas: 1
+
+DexApp:
```
Comment (coffeepac) [inline dex/values.yaml]: the DexApp seems to be a demonstration application for how to use Dex.  If this is the case, it shouldn't be a part of this Chart.  Charts in this repository should be consumable by any production user and a production user won't want a sample app.

I would include a link in the README for this chart to a guide that points back to a samsung-cnct repo (or your personal repo) with the sample dex-app and the DexApp chart and configurations.

File: dex/values.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+Dex:
+  Image: "quay.io/coreos/dex"
+  ImageTag: "v2.0.0-beta.1"
+  ImagePullPolicy: "Always"
+  Component: "dex"
+  NodePort: 30443
```
Comment (coffeepac) [inline dex/values.yaml]: is there a way to let this be auto-assigned by kubernetes?  manually assigning it has a small amount of risk for collisions.

**Comment by davidewatson** [comment]
Lgtm

File: dex/values.yaml (hunk 1 above)
Comment (keyolk) [inline dex/values.yaml]: self signed credentials 
doesnt matter~

File: dex/Chart.yaml (hunk 2 above)
Comment (keyolk) [inline dex/Chart.yaml]: OK I'll

File: dex/values.yaml (hunk 3 above)
Comment (keyolk) [inline dex/values.yaml]: and
so far, there is no auth interface ( like login ) in kube-apiserver.
the app is used for get Token for kube user.

if vault chart doesn't include the app.
we may have to add dex-app chart to our atlas or
use local chart repo for k2.

to make dex enabeld kube by k2.

File: dex/values.yaml (hunk 4 above)
Comment (coffeepac) [inline dex/values.yaml]: yes.

File: dex/values.yaml (hunk 1 above)
Comment (coffeepac) [inline dex/values.yaml]: okay

File: dex/values.yaml (hunk 3 above)
Comment (coffeepac) [inline dex/values.yaml]: ahh, so this app is required for dex to be functional within k7s.  this app should stay in this chart then.

File: dex/values.yaml (hunk 4 above)
Comment (keyolk) [inline dex/values.yaml]: access point(Whether use nodeport or others) must be same to some of configuration in this chart
like values in
Issuer, RedirectURI, ofcourse with Pki assets ...
Do you have any idea inform that to user ?

File: dex/values.yaml (hunk 3 above)
Comment (keyolk) [inline dex/values.yaml]: Then I'll let that in

File: dex/values.yaml (hunk 4 above)
Comment (coffeepac) [inline dex/values.yaml]: I do not.  this needs to be hardcoded for now then.

File: dex/Chart.yaml (hunk 2 above)
Comment (coffeepac) [inline dex/Chart.yaml]: I'll merge this for now, but if you get a chance, please add additional keywords as appropriate

**Comment by coffeepac** [comment]
LGTM

**Pull request merged**